cli.sh

```diff
@@ -750,6 +759,70 @@ execute_che_dir() {
   docker_run_with_che_properties -v "$CURRENT_DIRECTORY":"$CURRENT_DIRECTORY" "${CHE_DIR_IMAGE_NAME}":"${CHE_UTILITY_VERSION}" "${CURRENT_DIRECTORY}" "$@"
 }
 
+execute_che_ws() {
+  debug $FUNCNAME
+  if [ -z ${1+x} ]; then 
+    error "${CHE_PRODUCT_NAME} (che ws): No command provided."
+    return
+  fi
+
+  case $1 in
+    list)
+      check_current_image_and_update_if_not_found ${CHE_ACTION_IMAGE_NAME} ${CHE_UTILITY_VERSION}
+      docker_run_with_che_properties "${CHE_ACTION_IMAGE_NAME}":"${CHE_UTILITY_VERSION}" list-workspaces
+    ;;
+    logs)
+      if [ $# -eq 2 ]; then
+        info "${CHE_PRODUCT_NAME} (che ws): Connecting to workspace with id ${2}"
+        WS_ID=${2}
+
+        # ID not provided, let's do a simple discovery of running workspaces
+      else 
+        info "${CHE_PRODUCT_NAME} (che ws): Searching for running workspaces..."
+
+        CURRENT_WS_INSTANCES=$(docker ps -aq --filter "name=workspace")
+        CURRENT_WS_COUNT=$(echo $CURRENT_WS_INSTANCES | wc -w)
+      
+        # No running workspaces
+        if [ $CURRENT_WS_COUNT -eq 0 ]; then
+          error "${CHE_PRODUCT_NAME} (che ws): We could not find any running workspaces"
+          return
+
+        # Exactly 1 running workspace
+        elif [ $CURRENT_WS_COUNT -eq 1 ]; then
+
+          WS_ID=$CURRENT_WS_INSTANCES
+          info "${CHE_PRODUCT_NAME} (che ws): Connecting to workspace with id ${WS_ID}"
+
+        # 2+ running workspace
+        else 
+          info "${CHE_PRODUCT_NAME} (che ws): Re-run with 'che ws <ws-id>'"
+          IFS=$'\n'
+
+          echo "WS CONTAINER ID    HAS SSH?    SSH PORT"
+          for CHE_WS_CONTAINER_ID in $CURRENT_WS_INSTANCES; do
+            CURRENT_WS_PORT=""
+            if has_ssh ${CHE_WS_CONTAINER_ID}; then 
+              CURRENT_WS_PORT=$(docker inspect --format='{{ (index (index .NetworkSettings.Ports "22/tcp") 0).HostPort }}' ${CHE_WS_CONTAINER_ID})
+            fi
+            echo "$CHE_WS_CONTAINER_ID       $(has_ssh ${CHE_WS_CONTAINER_ID} && echo "y" || echo "n")           $CURRENT_WS_PORT"
+          done
+          return
+        fi
+      fi
+
+      info "Displaying logs for container $WS_ID"
+      docker_exec exec ${WS_ID} find /home/ -iname "catalina-0.log" -exec tail -n200 {} \;
```

Comment:
Right now - it only grabs a single machine set of logs. If there are multiple machines that are running, then it asks the user to choose which machine that they should connect to.  One of the improvements that I could make is to scan all of the machines that are running, and only list the workspace machines that are running a dev agent on it, and then allow for connection into that.